plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -0,0 +1,391 @@
+            assertUpdate("INSERT INTO " + testTable.getName() + " (one, three) VALUES (123, 'test')", 1);
+            assertThatThrownBy(() -> query(format("SELECT * FROM TABLE(system.query(query => 'SELECT * FROM %s'))", testTable.getName())))
+                    .hasMessage("line 1:21: Table function system.query not registered");
+        }
+    }
+
+    @Override
+    public void testNativeQueryCreateStatement()
+    {
+        // table function disabled for Ignite, because it doesn't provide ResultSetMetaData, so the result relation type cannot be determined
+        assertFalse(getQueryRunner().tableExists(getSession(), "numbers"));
+        assertThatThrownBy(() -> query("SELECT * FROM TABLE(system.query(query => 'CREATE TABLE numbers(n INTEGER)'))"))
+                .hasMessage("line 1:21: Table function system.query not registered");
+        assertFalse(getQueryRunner().tableExists(getSession(), "numbers"));
+    }
+
+    @Override
+    public void testNativeQueryInsertStatementTableDoesNotExist()
+    {
+        // table function disabled for Ignite, because it doesn't provide ResultSetMetaData, so the result relation type cannot be determined
+        assertFalse(getQueryRunner().tableExists(getSession(), "non_existent_table"));
+        assertThatThrownBy(() -> query("SELECT * FROM TABLE(system.query(query => 'INSERT INTO non_existent_table VALUES (1)'))"))
+                .hasMessage("line 1:21: Table function system.query not registered");
+    }
+
+    @Override
+    public void testNativeQueryInsertStatementTableExists()
+    {
+        // table function disabled for Ignite, because it doesn't provide ResultSetMetaData, so the result relation type cannot be determined
+        try (TestTable testTable = simpleTable()) {
+            assertThatThrownBy(() -> query(format("SELECT * FROM TABLE(system.query(query => 'INSERT INTO %s VALUES (3, 4)'))", testTable.getName())))
+                    .hasMessage("line 1:21: Table function system.query not registered");
+            assertQuery("SELECT * FROM " + testTable.getName(), "VALUES (1, 1), (2, 2)");
+        }
+    }
+
+    @Override
+    protected TestTable simpleTable()
+    {
+        return new TestTable(onRemoteDatabase(), format("%s.simple_table", getSession().getSchema().orElseThrow()), "(col BIGINT, id bigint primary key)", ImmutableList.of("1, 1", "2, 2"));
+    }
+
+    @Override
+    public void testNativeQueryIncorrectSyntax()
+    {
+        // table function disabled for Ignite, because it doesn't provide ResultSetMetaData, so the result relation type cannot be determined
+        assertThatThrownBy(() -> query("SELECT * FROM TABLE(system.query(query => 'some wrong syntax'))"))
+                .hasMessage("line 1:21: Table function system.query not registered");
+    }
+
+    @Override
+    protected void verifyAddNotNullColumnToNonEmptyTableFailurePermissible(Throwable e)
+    {
+        // Ignite will not throw any exception here.
```

Comment:
This looks like a bug. Does Ignite have a jira? Could you file an issue to https://github.com/trinodb/trino/issues/new?

Reply:
Add not null column to the non-empty table, Ignite not throw exception and does not give default value.
I think it is the Ignite bug, just created one [issue](https://issues.apache.org/jira/browse/IGNITE-18829) to Ignite.

Adjusted the comment.

Reply:
Thanks for filing the issue. Additionally, we can deny `NOT NULL` option when adding a new column in this PR.

Reply:
Is it ok to keep it here, and once the Ignite issue fixed we can add the exception check?

Reply:
I don't think so. Please deny `NOT NULL` option when adding a new column.

Reply:
Ok, how about deny the `add column` option in this PR?

Reply:
Do you mean denying the entire `ADD COLUMN`? It's also fine.